src/schema/comments.ts

```diff
@@ -333,9 +364,24 @@ const getMentions = async (
     return [];
   }
 
-  return con
+  const users = await con
     .getRepository(User)
     .find({ where: { username: In(result), id: Not(userId) } });
+
+  if (!sourceId) {
+    return users;
+  }
+
+  const source = await con.getRepository(Source).findOneBy({ id: sourceId });
+
+  if (!source.private) {
+    return users;
+  }
+
+  const ids = users.map(({ id }) => id);
+  const members = await getMemberIds(con, sourceId, ids);
+
+  return users.filter(({ id }) => members.includes(id));
```

Comment:
It's better to do this within the database. select members that are in the provided array. Otherwise there could be squads with hundreds of members

Reply:
We are doing that already in the `getMemberIds`

Reply:
Oh I see, but still it's not needed as it can be basically turned into one sql statement that returns the users already filtered

Reply:
This only happens when you have `sourceId` in your parameters. If none, it is just the user id. Though we can definitely combine them and delete the `getMemberIds` function.

Reply:
Yes. I think this needs to be fixed